### ISSUE TITLE
When detecting a country block, relay the download to a proxied youtube-dl executable. (This means we implement a second instance, yes)

### ISSUE THREAD

**Issue by GlassedSilver** [Tzahi12345/YoutubeDL-Material#230]
Hey,

so since when I get country blocks on the desktop I just turn on a VPN (since that's what I have setup anyways, but I think just proxying is also possible?) to watch a video, I think in an application that's usually working in the background without constant supervision it'd be very handy to add a proxied instance of youtube-dl to the set of fallback tools and techniques so less and less common youtube-dl errors are a problem and they are treated with less user interaction.

What do you think?

Cheers!

**Comment by Tzahi12345** [comment]
Hm there might be a simpler solution that running multiple instances. There's a `--proxy` arg ([see here](https://github.com/ytdl-org/youtube-dl#network-options)) which we could leverage. A setting like "on block use proxy" with a text input for the actual proxy, so blocked videos could get re-routed through the proxy if it fails due to georestrictions.

Definitely not a bad idea, wouldn't take much time to implement either!

**Comment by GlassedSilver** [comment]
I just noticed, this could also be helpful to go through a predefined list of proxies to use not just for geo blocks but also when we get the dreaded error 429 ("too many requests").

**Comment by Tzahi12345** [comment]
Absolutely, the only potential issue here is that people seem to have problems with the proxy arg [from time to time](https://github.com/Tzahi12345/YoutubeDL-Material/issues/177#issuecomment-730450463). If it works just fine for you, then I assume it's not widespread and I'll gladly implement this feature :)

**Comment by GlassedSilver** [comment]
Maybe if the ytdl proxy command is the thing that causes issues we could implement an alternative to use ytdl proxied?

Didn't use the proxy arg yet, because with the one off downloads so far everything had been smooth.
